### ISSUE TITLE
Cannot train cut_model with input grayscale images (opt.input_nc==1)

### ISSUE THREAD

**Issue by adrienchaton** [taesungp/contrastive-unpaired-translation#110]
Hi and thanks for this great repo,

I would like to train AtoB where A are grayscale images and B are RGB images.
I tried specifically saving A images as grayscale (e.g. using img = PIL.Image.open(file).convert('L').save(file) in batch ) in the trainA folder. And in the train.py option I pass --input_nc 1 however this creates errors such as:

```
Traceback (most recent call last):
  File "train.py", line 43, in <module>
    model.data_dependent_initialize(data)
  File "/fast-2/adrien/pix2pix/contrastive-unpaired-translation-master/models/cut_model.py", line 117, in data_dependent_initialize
    self.compute_G_loss().backward()                   # calculate graidents for G
  File "/fast-2/adrien/pix2pix/contrastive-unpaired-translation-master/models/cut_model.py", line 200, in compute_G_loss
    self.loss_NCE = self.calculate_NCE_loss(self.real_A, self.fake_B)
  File "/fast-2/adrien/pix2pix/contrastive-unpaired-translation-master/models/cut_model.py", line 215, in calculate_NCE_loss
    feat_q = self.netG(tgt, self.nce_layers, encode_only=True)
  File "/fast-2/adrien/pix2pix/venv_tmp/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/fast-2/adrien/pix2pix/contrastive-unpaired-translation-master/models/networks.py", line 994, in forward
    feat = layer(feat)
  File "/fast-2/adrien/pix2pix/venv_tmp/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/fast-2/adrien/pix2pix/venv_tmp/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/fast-2/adrien/pix2pix/venv_tmp/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 396, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: Given groups=1, weight of size [64, 1, 7, 7], expected input[1, 3, 518, 518] to have 1 channels, but got 3 channels instead
```

Could anyone fix this and train a gray to RGB image model please ?

**Comment by taesungp** [comment]
Hi @adrienchaton, that is actually not possible with CUT. It is because the same encoder needs to be used for both domains. Therefore, they can be both grayscale, or both RGB, but not mixed.

I think the simplest way to work around this is converting the grayscale images into RGB images with equal color intensities.

**Comment by adrienchaton** [comment]
Got it ! Thanks @taesungp

**Comment by edwardcho** [comment]
Hello Sir.

I tested CUT model using 3ch image (like grayscale).
But I got strange results.

This is my raw-data. (grayscale) : real_A
![1](https://user-images.githubusercontent.com/29569662/147037295-0abb5f6e-2064-42e7-aad4-8df9d2b60de2.png)

This is result-data for my raw-data : fake_A
![1](https://user-images.githubusercontent.com/29569662/147037354-9a9550fe-7034-4777-952c-47ac2fb49357.png)

Maybe, This problem was occured channels problem of input-image. Right??

How to solve it? (I want to get grayimage as results.)

Thanks.
Edward Cho.
